src/main/java/mate/acadamy/internetshop/dao/impl/BucketDaoImpl.java

```diff
@@ -0,0 +1,59 @@
+package mate.acadamy.internetshop.dao.impl;
+
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import mate.acadamy.internetshop.dao.BucketDao;
+import mate.acadamy.internetshop.db.Storage;
+import mate.acadamy.internetshop.lib.Dao;
+import mate.acadamy.internetshop.model.Bucket;
+
+@Dao
+public class BucketDaoImpl implements BucketDao {
+    @Override
+    public Bucket create(Bucket bucket) {
+        Optional<Bucket> currentBucket = Storage.buckets
```

Comment:
we can use `set` if we want to save only origin element

Reply:
What do you mean? we use ```set```  in ```update``` method

Reply:
якщо я правильно зрозуміла то в цьому методі ти перевіряєш чи існує уже такий елемент в списку, щоб не додавати його двічі. В такому випадку ми можемо просто використати сет для зберігання даних, і тоді не доведеться перевіряти чи існує такий же елемент в середині

Reply:
But we can`t add a new element with ```set```, only set by index in current ArrayList length, and it will be duplicate of method update. We can do something like this: 
```........................................
if (!currentBucket.equals(Optional.of(bucket))) {
            Storage.buckets.add(bucket);
            return bucket;
        } else {
            update(bucket);
            return bucket;
        }```